ha/util/consul_kv_store.py

```diff
@@ -154,13 +197,31 @@ def get(self, key: str = ""):
         Return:
             str: Return dictionary of all key val pair.
         """
-        _, data = self._consul.kv.get(self._prepare_key(key), recurse=True)
-        if data is None:
-            return data
+        k = self._prepare_key(key)
+
+        # simulating consul kv get operation for _payload dict
+        payload_data = None
+        if self._enable_batch_put:
+            # Using dictionary comprehension + startswith()
+            # Prefix key match in dictionary
+            Log.debug(f"[consul-op] searching key in cache: {k}")
+            payload_data = {key:val for key, val in self._payload.items() if key.startswith(k)}
+
+        Log.debug(f"[consul-op] getting the value for key: {k}")
+        _, data = self._consul.kv.get(k, recurse=True)
+        Log.debug(f"[consul-op] got the value for Key: {k}")
+
         key_val: dict = {}
-        for key in data:
-            key_val[key['Key']] = key['Value'].decode("utf-8") if isinstance(key['Value'], bytes) else key['Value']
-        return key_val
+        if data:
+            for key in data:
+                key_val[key['Key']] = key['Value'].decode("utf-8") if isinstance(key['Value'], bytes) else key['Value']
+
+        # merge and/or overwrite the values from _payload with received data from consul kv store
```

Comment:
Why merge ? when we update KV in consul are we doing merge or overwriting the value ?

Reply:
no, in update and set both I am setting the value in cache only but '_, data = self._consul.kv.get(k, recurse=True)' fetches all the values recursively considering the input key as a prefix. so new values for some keys are updated in the cache but some do not have a new value in the cache so the value for these keys should be taken from the consul.
I merging and overwriting as some keys are updated/added in the cache needs to merge.
if you still have queries regarding this we can discuss